### PR TITLE
GenericSignatureImpl: Assert we were given type parameters in getSuperclassBound & conformsToProtocol

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
Follow-up to #31515

### CODE REVIEW

File: lib/AST/GenericSignature.cpp [hunk 1]
```diff
@@ -456,8 +457,7 @@ GenericSignatureImpl::getConformsTo(Type type) {
 }
 
 bool GenericSignatureImpl::conformsToProtocol(Type type, ProtocolDecl *proto) {
```
Comment (AnthonyLatsis) [inline lib/AST/GenericSignature.cpp:459]: This method is crying to be renamed to `requiresProtocol` despite the FIXIT it contains.

**Comment by AnthonyLatsis** [comment]
@swift-ci please test

**Comment by AnthonyLatsis** [comment]
@swift-ci please test source compatibility

**Comment by AnthonyLatsis** [comment]
@swift-ci please test source compatibility

**Comment by AnthonyLatsis** [comment]
@swift-ci please test source compatibility

**Review by CodaFi — APPROVED** [review]

**Comment by AnthonyLatsis** [comment]
🍄

**Pull request merged**